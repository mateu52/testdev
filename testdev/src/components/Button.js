import React from 'react';


export default class Button extends React.Component {
    state = { color: 'blue', counter: 0}
    handleClick = () => {
      this.setState({counter: this.state.counter +1})
    }
    render(){
      return <button onClick={this.handleClick}>
                {this.props.label} {this.state.counter}
              </button>;
    }
  }
  Button.defaultProps ={
    label: 'OK'
  }