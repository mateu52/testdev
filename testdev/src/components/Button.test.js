import React from "react";
import renderer from 'react-test-renderer';
import Button from './Button';

it('renders ok', () => {
    const tree = renderer
        .create(<Button label="Click me" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});                

describe('testing with jest', () => {
    it('should isOK to by true', () => {
        const isOK = true;
        expect(isOK).toBe(true);
    });
    it('should array length to be 2', () => {
        const names = ['Patryk' , 'Ania'];
        expect(names.length).toBe(2);
     });
});