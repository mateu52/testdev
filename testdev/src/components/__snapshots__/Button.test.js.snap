// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<button
  onClick={[Function]}
>
  Click me
   
  0
</button>
`;
